packages/hw-app-ckb/src/Ckb.js

```diff
@@ -0,0 +1,178 @@
+//@flow
+
+import type Transport from "@ledgerhq/hw-transport";
+import BIPPath from "bip32-path";
+
+/**
+ * Nervos API
+ *
+ * @example
+ * import Ckb from "@ledgerhq/hw-app-ckb";
+ * const ckb = new Ckb(transport);
+ */
+export default class Ckb {
+  transport: Transport<*>;
+
+  constructor(transport: Transport<*>, scrambleKey: string = "CKB") {
+    this.transport = transport;
+    transport.decorateAppAPIMethods(
+      this,
+      [
+        "getAppConfiguration",
+        "getWalletId",
+        "getWalletPublicKey",
+        "signTransaction"
+      ],
+      scrambleKey
+    );
+  }
+
+  /**
+   * get CKB address for a given BIP 32 path.
+   *
+   * @param path a path in BIP 32 format
+   * @return an object with a publicKey
+   * @example
+   * const result = await ckb.getWalletPublicKey("44'/144'/0'/0/0");
+   * const publicKey = result;
+   */
+  async getWalletPublicKey(path: string): Promise<string> {
+    const bipPath = BIPPath.fromString(path).toPathArray();
+
+    const cla = 0x80;
+    const ins = 0x02;
+    const p1 = 0x00;
+    const p2 = 0x00;
+    const data = Buffer.alloc(1 + bipPath.length * 4);
+
+    data.writeUInt8(bipPath.length, 0);
+    bipPath.forEach((segment, index) => {
+      data.writeUInt32BE(segment, 1 + index * 4);
+    });
+
+    const response = await this.transport.send(cla, ins, p1, p2, data);
+    const publicKeyLength = response[0];
+    return response.slice(1, 1 + publicKeyLength).toString("hex");
+  }
+
+  /**
+   * Sign a Nervos transaction with a given BIP 32 path
+   *
+   * @param path a path in BIP 32 format
+   * @param rawTxHex transaction to sign
+   * @param contextTransaction transaction context to use in parsing (optional)
+   * @return a signature as hex string
+   * @example
+   * const signature = await ckb.signTransaction("44'/144'/0'/0/0", "00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000");
+   */
+  async signTransaction(
+    path: string,
+    rawTxHex: string,
+    rawContextTxHex: string
```

Comment:
I think this needs to be an array; we have one context transaction per input in the current transaction.